File: PaSho Tracker/Domain/Model/TaskModel.cs
using System.Text.Json.Serialization;
using PaSho_Tracker.Enums;
using TaskStatus = PaSho_Tracker.Enums.TaskStatus;

namespace PaSho_Tracker.Model;

public class TaskModel : BaseEntity
{
    public TaskModel(int id, string title, string description, TaskPriority priority, TaskStatus status,
        DateTime deadline, string assignedUserId)
        : base(id)
    {
        Title = title;
        Description = description;
        Priority = priority;
        Status = status;
        Deadline = deadline;
        AssignedUserId = assignedUserId;
        Comments = new List<CommentModel>();
    }
    
    public TaskModel()
    {
        Comments = new List<CommentModel>();
    }

    public string Title { get; set; }
    public string Description { get; set; }

    public TaskPriority Priority { get; set; }
    public TaskStatus Status { get; set; }

    public DateTime Deadline { get; set; }
    public string AssignedUserId { get; set; }
    [JsonIgnore] public ICollection<CommentModel> Comments { get; set; }

    public override string GetEntityInfo()
    {
        return $"Title = {Title}, Description = {Description}";
    }
}

File: frontend/src/types/task-model.ts
export type TaskStatus = 'ToDo' | 'InProgress' | 'Done'

export type TaskPriority = 'Low' | 'Medium' | 'High'

export interface CommentModel {
  id: number
  commentText: string
  author: string
  relatedTaskId: number
}

export interface TaskModel {
  id: number
  title: string
  description: string
  priority: TaskPriority
  status: TaskStatus
  deadline: string // ISO string
  assignedUserId: string
  comments: CommentModel[]
  categoryId: number
}

export interface CategoryModel {
  id: number
  categoryName: string
}
